carbonmark-api/src/utils/marketplace.utils.ts

```diff
@@ -0,0 +1,33 @@
+import { merge } from "lodash";
+import { Listing } from "../.generated/types/marketplace.types";
+import { FirebaseInstance } from "../plugins/firebase";
+import { getFirebaseUser } from "./firebase.utils";
+import { notNil } from "./functional.utils";
+
+export const updateListingUser =
+  (fb: FirebaseInstance) => async (listing: Partial<Listing>) => {
```

Comment:
I hear you but I think the singleton approach might go against Fastify's encapsulation principal (https://fastify.dev/docs/latest/Reference/Encapsulation).

i.e. if we created a seperate Fastify instance it would share the same Firebase instance rather than a seperate instance. Not sure if this is an issue but seems cleaner to have a new one per instance? (especially with versioning coming up)

What do you think?